### PR TITLE
batches: support multiple target branches

### PR DESCRIPTION
**Pull request by LawnGnome**
The client side part of sourcegraph/sourcegraph#25228. Depends on https://github.com/sourcegraph/sourcegraph/pull/28168, and will not pass CI until that's merged and this PR gets an update to its `go.mod`.

### CODE REVIEW

File: internal/batches/service/workspaces.go
Comment (LawnGnome) [inline internal/batches/service/workspaces.go:41]: Fun fact: `repoByID` was never used again.

File: internal/batches/service/service.go
Comment (LawnGnome) [inline internal/batches/service/service.go:449]: This is obviously eerily similar to code in https://github.com/sourcegraph/sourcegraph/pull/28168 — my not-so-secret plan here is to unify these once we have generics available in Go, since the only real distinction is that the "revision" type is different in `src` and SSBC.

**Comment by LawnGnome** [comment]
I'm opening this up for early review, but note the... uh, note in the PR text about this not being able to pass until the corresponding Sourcegraph PR lands.

**Review by mrnugget — APPROVED** [review]

File: internal/batches/service/workspaces.go
Comment (eseliger) [reply internal/batches/service/workspaces.go:41]: Where you at golangci-lint when we need you 🧐

**Review by eseliger — APPROVED** [review]
In general LGTM, pending my comment on the algorithm on the main repo PR.

**Pull request merged**
